### ISSUE TITLE
Cookie auth mode hard coded 

### ISSUE THREAD

**Issue by trivialkettle** [supertokens/supertokens-flutter#42]
Hi,
Any reason why the cookie based auth is hard coded at?
https://github.com/supertokens/supertokens-flutter/blob/3182637321f8213c6109839bd276cd34a4469589/lib/src/supertokens-http-client.dart#L217

If I change it to match https://github.com/supertokens/supertokens-flutter/blob/3182637321f8213c6109839bd276cd34a4469589/lib/src/supertokens-http-client.dart#L85

The header based auth forced by my backend works.

**Comment by rishabhpoddar** [comment]
This seems to be a bug. We will fix it asap.

**Comment by nkshah2** [comment]
Hi @trivialkettle 

Can you post the config on both the frontend and backend you were using initially? Because even with the hardcoded header it should have worked fine for you

**Comment by trivialkettle** [comment]
Frontend (flutter):
```
 SuperTokens.init(
      apiDomain: "http://localhost:3001",
      apiBasePath: 'auth',
      tokenTransferMethod: SuperTokensTokenTransferMethod.HEADER);
```

Backend (go):
```
return supertokens.TypeInput{
		Supertokens: &supertokens.ConnectionInfo{
			ConnectionURI: config.Core,
			APIKey:        config.Apikey,
		},
		AppInfo: supertokens.AppInfo{
			AppName:       appName,
			APIDomain:     config.Api,
			WebsiteDomain: config.Website,
		},
		RecipeList: []supertokens.Recipe{
			session.Init(&sessmodels.TypeInput{
				GetTokenTransferMethod: func(req *http.Request, forCreateNewSession bool, userContext supertokens.UserContext) sessmodels.TokenTransferMethod {
					return sessmodels.HeaderTransferMethod
				},
			}),
		},
	}
```

I hardcoded the token transfer mode to header based. I read the warnings and infos at https://supertokens.com/docs/emailpassword/common-customizations/sessions/token-transfer-method, but had to use it to do some tests with `curl`.

**Comment by rishabhpoddar** [comment]
> Edit: I faced the issue in attemptRefreshingSession()

What issue did you face? Cause according to us, it should have just worked regardless of the hard coding.

**Comment by trivialkettle** [comment]
`attemptRefreshingSession()` returned `false`.

**Comment by rishabhpoddar** [comment]
We can try and replicate this issue. Will get back

**Comment by nkshah2** [comment]
Hey @trivialkettle 

This has been fixed in version `0.2.8`. Closing this for now, feel free to re-open if the issue persists
